qfieldsync/gui/cloud_converter_dialog.py

```diff
@@ -0,0 +1,247 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+ QFieldCloudConverterDialog
+                                 A QGIS plugin
+ Sync your projects to QField on android
+                             -------------------
+        begin                : 2021-057-22
+        git sha              : $Format:%H$
+        copyright            : (C) 2015 by OPENGIS.ch
+        email                : info@opengis.ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+import os
+import re
+from pathlib import Path
+
+from qgis.core import Qgis, QgsProject, QgsProviderRegistry
+from qgis.PyQt.QtCore import Qt
+from qgis.PyQt.QtWidgets import QApplication, QDialog, QDialogButtonBox, QMessageBox
+from qgis.PyQt.uic import loadUiType
+
+from qfieldsync.core.cloud_api import CloudException
+from qfieldsync.core.cloud_converter import CloudConverter
+from qfieldsync.core.cloud_project import CloudProject
+from qfieldsync.core.cloud_transferrer import CloudTransferrer
+from qfieldsync.core.preferences import Preferences
+from qfieldsync.gui.cloud_login_dialog import CloudLoginDialog
+from qfieldsync.libqfieldsync.utils.file_utils import fileparts
+
+from ..utils.qt_utils import make_folder_selector
+
+DialogUi, _ = loadUiType(
+    os.path.join(os.path.dirname(__file__), "../ui/cloud_converter_dialog.ui")
+)
+
+
+class CloudConverterDialog(QDialog, DialogUi):
+    def __init__(self, iface, network_manager, project, parent=None):
+        """Constructor."""
+        super(CloudConverterDialog, self).__init__(parent=parent)
+        self.setupUi(self)
+
+        self.iface = iface
+        self.project = project
+        self.qfield_preferences = Preferences()
+        self.network_manager = network_manager
+
+        if not self.network_manager.has_token():
+            login_dlg = CloudLoginDialog(self.network_manager, self)
+            login_dlg.authenticate()
+            login_dlg.accepted.connect(
+                lambda: self.network_manager.projects_cache.refresh()
+            )
+            login_dlg.rejected.connect(lambda: self.close())
+        else:
+            self.network_manager.projects_cache.refresh()
+
+        project_name = self.project.baseName()
+        if project_name:
+            pattern = re.compile("[\W_]+")  # NOQA
```

Comment:
```suggestion
            pattern = re.compile(r"[\W_]+")
```

Or is this really meant to be this way?